src/analysis/retail/monk/windwalker/modules/spells/FistsofFury.tsx

```diff
@@ -41,16 +48,23 @@ class FistsofFury extends Analyzer {
     if (!this.isNewFistsTick(event.timestamp)) {
       return;
     }
-    this.fistsTicks += 1;
+    if (!this.selectedCombatant.hasBuff(TALENTS_MONK.SERENITY_TALENT.id) && !this.inSerenity) {
+      this.nonSerenityFistsTicks += 1;
+    }
     this.previousTickTimestamp = event.timestamp;
   }
 
-  get averageTicks() {
-    return this.fistsTicks / this.casts;
+  onFistsCast(event: CastEvent) {
+    if (!this.selectedCombatant.hasBuff(TALENTS_MONK.SERENITY_TALENT.id)) {
+      this.inSerenity = false;
+      this.nonSerenityCasts += 1;
+    } else {
+      this.inSerenity = true;
```

Comment:
This seems like an odd way to be tracking whether serenity is active or not, I think I would rather see event listeners tied to the buff  apply/remove events of serenity itself or inject the serenity module if there is tracking for its active state there

Reply:
Ok, I'll inject the serenity module. Is there a way I could use this module in an apl condition as well?

Reply:
I renamed this to lastCastInSerenity, which is a better name.
The idea is I don't want to count any damage ticks from a FoF cast made during serenity (even if the serenity buff expires), since we we aren't counting that cast in the denominator of the final calculation.